db/crud.go

```diff
@@ -1165,6 +1165,93 @@ func (db *DatabaseCollectionWithUser) PutExistingRevWithBody(ctx context.Context
 
 }
 
+func (db *DatabaseCollectionWithUser) SyncFnDryrun(ctx context.Context, body Body, docID string) (*uint32, base.Set, channels.AccessMap, channels.AccessMap, error) {
+	doc := &Document{
+		ID:    docID,
+		_body: body,
+	}
+	oldDoc := doc
```

Comment:
This looks like if a body is passed in and docID is not found, we're using the body as both doc and oldDoc.  I don't think this is what users will want to do - if they are only passing body, I think it should behave as an insert (oldDoc=null)

Reply:
Its treated as an insert if there is no docID, but olddoc is used in prepareSyncFn. If doc id is not found it returns 404